gtsam/hybrid/HybridBayesTree.cpp

```diff
@@ -41,6 +42,22 @@ bool HybridBayesTree::equals(const This& other, double tol) const {
   return Base::equals(other, tol);
 }
 
+/* ************************************************************************* */
+DiscreteValues HybridBayesTree::discreteMaxProduct(
```

Comment:
This should be in DiscreteFactorGraph?

Reply:
This is special behavior leveraging the `TableFactor`. I don't think we should mess with `DiscreteFactorGraph` wrt this.